packages/components/src/Provider/MarigoldProvider.tsx

```diff
@@ -1,15 +1,33 @@
 import React from 'react';
 import { OverlayProvider } from '@react-aria/overlays';
-import { ThemeProvider, ThemeProviderProps } from '@marigold/system';
+import {
+  Theme,
+  ThemeProvider,
+  ThemeProviderProps,
+  useTheme,
+} from '@marigold/system';
+
+export interface ThemeContextValue {
+  theme: Theme;
+}
+
+export const defaultThemeValue: ThemeContextValue = {
+  theme: {},
+};
 
 // a merge of the ThemeProvider and the react-aria OverlayProvider
 export const MarigoldProvider: React.FC<ThemeProviderProps> = ({
   theme,
   children,
 }) => {
+  const outerTheme = useTheme();
+  const outerThemeJson = JSON.stringify(outerTheme, null, 2);
+  const defaultThemeValueJson = JSON.stringify(defaultThemeValue, null, 2);
+  const isTopLevel = outerThemeJson === defaultThemeValueJson;
```

Comment:
Sorry, but that's not the proper way 🙈 Why can you check with the returned value from the theme hook?

Reply:
Hey @sebald I think now is the check correct but I am not sure. 
And we don't know how to check in jest if there are not more than one OverlayProvider. Do you have a tip for this?